features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/des/des_alt.c

```diff
@@ -0,0 +1,410 @@
+}
+#endif /* MBEDTLS_CIPHER_MODE_CBC */
+
+/*
+ * 3DES-ECB block encryption/decryption
+ */
+int mbedtls_des3_crypt_ecb( mbedtls_des3_context *ctx,
+                     const unsigned char input[8],
+                     unsigned char output[8] )
+{
+    unsigned char iv[8] = {0x00};
+    return mbedtls_des_docrypt(ctx->keyopt, ctx->key, ctx->enc, TDES_MODE_ECB, 8, iv, input, output);
+}
+
+#if defined(MBEDTLS_CIPHER_MODE_CBC)
+/*
+ * 3DES-CBC buffer encryption/decryption
+ */
+int mbedtls_des3_crypt_cbc( mbedtls_des3_context *ctx,
+                     int mode,
+                     size_t length,
+                     unsigned char iv[8],
+                     const unsigned char *input,
+                     unsigned char *output )
+{
+    return mbedtls_des_docrypt(ctx->keyopt, ctx->key, mode == MBEDTLS_DES_ENCRYPT, TDES_MODE_CBC, length, iv, input, output);
+}
+#endif /* MBEDTLS_CIPHER_MODE_CBC */
+
+
+
+static int mbedtls_des_docrypt(uint16_t keyopt, uint8_t key[3][MBEDTLS_DES_KEY_SIZE], int enc, uint32_t tdes_opmode, size_t length, 
+    unsigned char iv[8], const unsigned char *input, unsigned char *output)
+{
+    if (length % 8) {
+        return MBEDTLS_ERR_DES_INVALID_INPUT_LENGTH;
+    }
+    
+    // NOTE: Don't call driver function TDES_Open in BSP because it doesn't support TDES_CTL[3KEYS] setting.
+    CRPT->TDES_CTL = (0 << CRPT_TDES_CTL_CHANNEL_Pos) | (enc << CRPT_TDES_CTL_ENCRPT_Pos) |
+        tdes_opmode | (TDES_IN_OUT_WHL_SWAP << CRPT_TDES_CTL_BLKSWAP_Pos);
+
+    // Keying option 1: All three keys are independent.
+    // Keying option 2: K1 and K2 are independent, and K3 = K1.
+    // Keying option 3: All three keys are identical, i.e. K1 = K2 = K3. 
+    if (keyopt == 1) {
+        CRPT->TDES_CTL |= CRPT_TDES_CTL_3KEYS_Msk;
+    }
+    else {
+        CRPT->TDES_CTL &= ~CRPT_TDES_CTL_3KEYS_Msk;
+    }
+
+    // Set DES/TDES keys
+    // NOTE: Don't call driver function TDES_SetKey in BSP because it doesn't support endian swap.
+    uint32_t val;
+    volatile uint32_t *tdes_key = (uint32_t *) ((uint32_t) &CRPT->TDES0_KEY1H + (0x40 * 0));
+    val = nu_get32_be(key[0] + 0);
+    *tdes_key ++ = val;
+    val = nu_get32_be(key[0] + 4);
+    *tdes_key ++ = val;
+    val = nu_get32_be(key[1] + 0);
+    *tdes_key ++ = val;
+    val = nu_get32_be(key[1] + 4);
+    *tdes_key ++ = val;
+    val = nu_get32_be(key[2] + 0);
+    *tdes_key ++ = val;
+    val = nu_get32_be(key[2] + 4);
+    *tdes_key ++ = val;
+    
+    uint32_t rmn = length;
+    const unsigned char *in_pos = input;
+    unsigned char *out_pos = output;
+    
+    while (rmn) {
+        uint32_t data_len = (rmn <= MAXSIZE_DMABUF) ? rmn : MAXSIZE_DMABUF;
+        
+        uint32_t ivh, ivl;
+        ivh = nu_get32_be(iv);
+        ivl = nu_get32_be(iv + 4);
+        TDES_SetInitVect(0, ivh, ivl);
+    
+        memcpy(dmabuf_in, in_pos, data_len);
```

Comment:
Is this thread safe? `dmabuf_in` and `dmabuf_out` are static variables, which can be used by anyone calling this function